code/modules/food_and_drinks/machinery/coffeemaker.dm

```diff
@@ -408,3 +498,243 @@
 	. = ..()
 	atom_storage.max_slots = 4
 	atom_storage.set_holdable(list(/obj/item/coffee_cartridge))
+
+/*
+ * impressa coffee maker
+ * its supposed to be a premium line product, so its cargo-only, the board cant be therefore researched
+ */
+
+/obj/machinery/coffeemaker/impressa
+	name = "impressa coffeemaker"
+	desc = "An industry-grade Impressa Modello 5 Coffeemaker of the Piccionaia Home Appliances premium coffeemakers product line. Makes coffee from fresh dried whole beans."
+	icon = 'icons/obj/machines/coffeemaker.dmi'
+	icon_state = "coffeemaker_impressa"
+	circuit = /obj/item/circuitboard/machine/coffeemaker/impressa
+	initial_cartridge = null		//no cartridge, just coffee beans
+	brew_time = 15 SECONDS			//industrial grade, its faster than the regular one
+	density = TRUE
+	pass_flags = PASSTABLE
+	#define BEAN_CAPACITY 10	//amount of coffee beans that can fit inside
+	/// Current amount of coffee beans stored
+	var/coffee_amount = 0
+	/// List of coffee bean objects are stored
+	var/list/coffee = list()
+
+/obj/machinery/coffeemaker/impressa/Initialize(mapload)
+	. = ..()
+	if(mapload)
+		coffeepot = new /obj/item/reagent_containers/cup/coffeepot(src)
+		cartridge = null
+
+/obj/machinery/coffeemaker/impressa/Destroy()
+	QDEL_NULL(coffeepot)
+	QDEL_NULL(coffee)
+	return ..()
+
+/obj/machinery/coffeemaker/impressa/Exited(atom/movable/gone, direction)
+	if(gone == coffeepot)
+		coffeepot = null
+	if(gone == coffee)
+		coffee = null
+	return ..()
+
+/obj/machinery/coffeemaker/impressa/examine(mob/user)
+	. = ..()
+	if(coffee)
+		. += span_notice("The internal grinder conatins [coffee.len] [coffee.len == 1 ? "scoop" : "scoops"] of coffee beans")
+	return
```

Comment:
You do not need this

Reply:
Why not? The amount of beans left tells the user how many coffee pots can he brew

Reply:
The trailing return, you do not need it

Reply:
Done in [fa2fd29](https://github.com/tgstation/tgstation/pull/70991/commits/fa2fd2913571a945d618ce4441735d29692c5e0c)